JoinColony/colonyCDapp
# Crypto-to-fiat: Resolve edge cases with liquidation addresses

## Description

See issue #2793 for details.

This PR resolves a number of potential issues with liquidation addresses:

1. Check KYC handler was checking for existence of any liquidation address, instead of one linked to the user's bank account. If it doesn't exist, it will create one.
2. When updating bank account details, a new bank account is created
   1. For accounts with the same currency, the existing liquidation address needed to be updated with the new account ID.
   2. For accounts with different currencies, a new liquidation address should be created.
3. As there can be multiple liquidation addresses that can change at any point (primarily when user updates their bank details), we cannot rely on the ones stored in our DB. For that, a new query was added to get liquidation address for a given user.

It also removes the `bridgeXYZQuery` lambda, as handling both queries and mutations in a single lambda allows for less code duplication and easier deployments.

**Note:** In this PR I'm also testing the possibility of splitting amplify schema into multiple files in hope to make it more manageable, considering the existing schema is already nearly 4k lines long.

## Testing

Restart your dev env - otherwise the separate schemas will not get picked up.

Issue 1:

1. Make sure you set the Posthog and Persona env variables and run `npm run prepare`. 
2. Start KYC flow (you don't need to finish it though).
3. Add USD bank account details (EUR accounts can't be tested in sandbox). You will also need to use a US address.

On next refetch the `CheckKYCStatus` query should return a liquidation address.

Issue 2:

1. Update bank details to another USD account.
2. Check that the liquidation address is still the same.

In production, if you updated bank details to EUR account, new liquidation address should be created. In sandbox however we can't really test subpoint 2). 

Issue 3

You can run the following query with your user address to confirm it returns the liquidation address:
```gql
query LiqAdd {
  bridgeGetUserLiquidationAddress(userAddress: "")
}
```

Resolves #2793
